cmd/aries-agentd/main_test.go

```diff
@@ -2,280 +2,23 @@
-								"@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/invitation"
-							}`),
-				"application/didcomm-envelope-enc"),
-			expectedStatus:     http.StatusInternalServerError,
-			expectResponseData: false,
-		},
-	}
-	for _, tt := range tests {
-		resp, err := http.DefaultClient.Do(tt.r)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		defer func() {
-			e := resp.Body.Close()
-			if e != nil {
-				panic(err)
-			}
-		}()
-
-		require.Equal(t, tt.expectedStatus, resp.StatusCode)
-		if tt.expectResponseData {
-			require.NotEmpty(t, resp.Body)
-			response, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				t.Fatal(err)
-			}
-			require.NotEmpty(t, response)
-			require.True(t, isJSON(response))
-		}
-	}
-}
-
-// isJSON checks if response is json
-func isJSON(res []byte) bool {
-	var js map[string]interface{}
-	return json.Unmarshal(res, &js) == nil
-}
-
-func TestStartAriesDWithoutHost(t *testing.T) {
-	prev := os.Getenv(agentHostEnvKey)
-	defer func() {
-		err := os.Setenv(agentHostEnvKey, prev)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}()
-
-	err := os.Setenv(agentHostEnvKey, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	done := make(chan bool)
-
-	go func() {
-		main()
-		done <- true
-	}()
-
-	select {
-	case res := <-done:
-		require.True(t, res)
-	case <-time.After(5 * time.Second):
-		t.Fatal("agent should fail to start when host address not provided")
-	}
-}
-
-func TestStartAriesWithoutInboundHost(t *testing.T) {
-	prev := os.Getenv(agentHostEnvKey)
-	defer func() {
-		err := os.Setenv(agentHostEnvKey, prev)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}()
-
-	err := os.Setenv(agentHostEnvKey, randomURL())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	prevInboundHost := os.Getenv(agentHTTPInboundEnvKey)
-	defer func() {
-		inboundConfigErr := os.Setenv(agentHTTPInboundEnvKey, prevInboundHost)
-		if inboundConfigErr != nil {
-			t.Fatal(inboundConfigErr)
-		}
-	}()
-
-	err = os.Setenv(agentHTTPInboundEnvKey, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	done := make(chan bool)
-
-	go func() {
-		main()
-		done <- true
-	}()
-
-	select {
-	case res := <-done:
-		require.True(t, res)
-	case <-time.After(5 * time.Second):
-		t.Fatal("agent should fail to start when inbound host address not provided")
-	}
+// Correct behaviour is for main to finish with exit code 0.
+// This test fails otherwise.
+func TestCallingWithNoUserArgs(t *testing.T) {
+	setUpArgs()
+	main()
 }
 
-func generateTempDir(t testing.TB) (string, func()) {
-	path, err := ioutil.TempDir("", "db")
-	if err != nil {
-		t.Fatalf("Failed to create leveldb directory: %s", err)
-	}
-	return path, func() {
-		err := os.RemoveAll(path)
-		if err != nil {
-			t.Fatalf("Failed to clear leveldb directory: %s", err)
-		}
-	}
-}
+// Strips out the extra args that the unit test framework adds
+// This allows main() to execute as if it was called directly from the command line
+func setUpArgs() {
+	args := []string{os.Args[0]}
+	os.Args = args
```

Comment:
can be done in one line `os.Args = os.Args[:1]`

Reply:
Done!